packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -993,7 +993,7 @@ export const DatePickerMixin = (subclass) =>
             if (e.shiftKey) {
               this._overlayContent.focusCancel();
             } else {
-              this._overlayContent.focusDate(this._focusedDate);
+              this._overlayContent.focusDateElement();
```

Comment:
This fixes the issue with pressing <kbd>Tab</kbd> on the `input` when focused date is out of the viewport.